ExpenseReconciliation/Services/DashboardService.cs

```diff
@@ -45,5 +47,101 @@ public Task<IEnumerable<Split>> ListAsync()
         {
             return _splitRepository.ListAsync();
         }
-    }
+
+
+        public async Task<SplitSummary> SplitSummary(DateTime startDate, DateTime endDate, TimeUnit timeUnit)
+        {
+            var splitSummary = new SplitSummary();
+            var transactions = await _transactionRepository.GetByDateAsync(startDate,endDate) ;
+            var totalPeriod = GetTimeUnitStringList(startDate, endDate, timeUnit);
+            
+            var userAmountList = new List<Total>();
+            var unsplittedAmount=0m;
+            // summing amount
+            foreach (var tp in totalPeriod)
+            {
+                var userAmountDicPerPeriod = SumUsersAmount(transactions, tp);
+
+                foreach (var record in userAmountDicPerPeriod)
+                {
+                    if (record.Key != -1)
+                    {
+                        var total = new Total();
+                        total.userId = record.Key;
+                        total.amount = record.Value;
+                        total.timeUnit = tp;
+                        userAmountList.Add(total);
+                    }
+                    else
+                    {
+                        unsplittedAmount += record.Value;
+                    }
+                }
+            }
+
+            splitSummary.total = userAmountList;
+            splitSummary.startDate = startDate;
+            splitSummary.endDate = endDate;
+            splitSummary.unSplitted = unsplittedAmount;
+            splitSummary.timeunit = timeUnit.ToString();
+
+            return splitSummary;
+        }
+        
+        private List<string> GetTimeUnitStringList(DateTime startDate, DateTime endDate, TimeUnit timeUnit)
+        {
+            var timeUnitList = new List<string>();
+            if (timeUnit == TimeUnit.Month)
+            {
+                timeUnitList = Enumerable.Range(0, 13).Select(a => startDate.Date.AddMonths(a))
+                    .TakeWhile(a => a <= endDate.AddDays(DateTime.DaysInMonth(endDate.Year,endDate.Month)-endDate.Day))
+                    .Select(a => String.Concat(a.ToString("MMMM") + ", " + a.Year)).ToList();
+            }
+            else if (timeUnit == TimeUnit.Year)
+            {
+                timeUnitList = Enumerable.Range(0, 30).Select(a => startDate.Date.AddYears(a))
+                    .TakeWhile(a => a <= endDate)
+                    .Select(a => String.Concat( a.Year)).ToList();
+            }
+
+            return timeUnitList;
+        }
+
+        private Dictionary<int, decimal> SumUsersAmount(IEnumerable<Transaction> transactions, string timePeriodName)
```

Comment:
I find this function fairly hard to follow and it could be simplified, but if it works as is now, it can be added as is and refactored later.